### PR TITLE
Resolve various issues with UDFs

### PR DESCRIPTION
**Pull request by ptkool**
## What changes were proposed in this pull request?

Resolve a couple of issues with UDFs:

- Add missing nullable parameters
- Backport https://github.com/apache/spark/pull/18615

## How was this patch tested?

Ran Spark tests.

### CODE REVIEW

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -1919,18 +1919,39 @@ def __call__(self, *cols):
...
 def udf(f=None, returnType=StringType(), nullable=True):
     """Creates a :class:`Column` expression representing a user defined function (UDF).
-
     .. note:: The user-defined functions must be deterministic. Due to optimization,
         duplicate invocations may be eliminated or the function may even be invoked more times than
         it is present in the query.
-
     :param f: python function if used as a standalone function
     :param returnType: a :class:`pyspark.sql.types.DataType` object
-
```
Comment (JasonMWhite) [inline python/pyspark/sql/functions.py]: Let's eliminate any unnecessary whitespace diffs to keep it as close to stock as possible.

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -1919,18 +1919,39 @@ def __call__(self, *cols):
...
+        def wrapper(*args):
+            return self(*args)
+
+        wrapper.__name__ = self._name
+        wrapper.__module__ = (self.func.__module__ if hasattr(self.func, '__module__')
+                              else self.func.__class__.__module__)
+        wrapper.func = self.func
+        wrapper.returnType = self.returnType
+
+        return wrapper
```
Comment (JasonMWhite) [inline python/pyspark/sql/functions.py:1945]: 👍  Let's try to get this into 2.2.1

**Review by JasonMWhite — APPROVED** [review]

**Comment by JasonMWhite** [comment]
It feels a little dirty merging with no CI

**Review by angelini — APPROVED** [review]
